Sources/MapboxNavigation/SpriteRepository.swift

```diff
@@ -33,15 +33,22 @@ class SpriteRepository {
             completion()
             return
         }
-        updateSprite(styleURI: newStyleURI, completion: completion)
+        
+        spriteCache.clearMemory()
```

Comment:
Why not use `SpriteRepository.resetCache()`?

Reply:
Because the legacy shield icons won't be changed with the `Style`. Each `Style` has its own Sprite image and Sprite metadata, which are used to generate single Mapbox designed general shield icon. When the style changes, we only need to clear out the previous style related Sprite cache.